src/write.rs

```diff
@@ -0,0 +1,23 @@
+use std::{
+    ffi::CString,
+    io::{self, BufWriter, Seek, Write},
+    u32,
+};
+
+/// An extension for [std::io::Write]
+pub trait WriteExt: Write {
+    /// Read a little endian u32
+    fn write_u32_le(&mut self, value: u32) -> io::Result<()> {
+        self.write(&u32::to_le_bytes(value))?;
+        Ok(())
+    }
+
+    fn write_zeroes(&mut self, N: usize) -> io::Result<()> {
+        for _ in 0..N {
+            self.write(&[0; 0x1]);
+        }
+        Ok(())
+    }
```

Comment:
`N` should be lowercase because it's not a constant or a type.

This does not need to be a loop, calling write many times like this is inefficient; you could write a vec of zeroes instead. (This causes a memory allocation, but it's probably more efficient than calling write 20 times.)

This also ignores the error returned from `write` and always returns OK. (and also ignores the total amount of bytes written, but I suppose this is less important?)

```suggestion
    fn write_zeroes(&mut self, n: usize) -> io::Result<usize> {
        self.write(&vec![0; n])
    }
```